src/test/java/com/bravo/user/service/AddressServiceTest.java

```diff
@@ -0,0 +1,47 @@
+package com.bravo.user.service;
+
+import com.bravo.user.dao.model.Address;
+import com.bravo.user.dao.model.mapper.ResourceMapper;
+import com.bravo.user.dao.repository.AddressRepository;
+import com.bravo.user.model.dto.AddressDto;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
+class AddressServiceTest {
+    private AddressService testObj;
+    @Mock
+    private AddressRepository repository;
+    @Autowired
+    private ResourceMapper mapper;
+
+    @BeforeEach
+    void setUp() {
+        testObj = new AddressService(repository, mapper);
+    }
+
+    @Test
+    void findAllShouldCallRepositoryMethodAndMatchZip() {
+        Address address = new Address();
+        String zip = "test zip";
+        address.setZip(zip);
+        when(repository.findAll(any(Specification.class))).thenReturn(List.of(address));
+        List<AddressDto> results = testObj.findAllByUserId("");
+        assertEquals(zip, results.get(0).getZip());
+        verify(repository).findAll(any(Specification.class));
```

Comment:
You should not `verify` with concepts like `any()` since the implementation can become flawed and your test won't catch that (because you're mocking the `findAll`. Instead you should only use `any()` in your `when` statements and use the actual objects in your `verify` statements. If you do this then you can be less specific in your `when` statements.

     final String userId = "";
     // mock data     

     when(repository.findAll(any()).thenReturn(List.of(address));
   
     final List<AddressDto> results = addressService.findAllByUserId(userId);
     // asserts 

     final AddressFilter filter = new AddressFilter(Set.of(userId));
     final AddressSpecification specification = new AddressSpecification(filter);
     verify(repository).findAll(specification);

Reply:
Makes sense. Sometimes I tack it on the back of assertions just to make sure the method gets called but I guess it's not very useful in this case.